cmd/access/cluster.go

```diff
@@ -0,0 +1,446 @@
+func (c *clusterAccessOptions) Println(msg string) {
+	osdctlutil.StreamPrintln(c.IOStreams, msg)
+}
+
+// Print prints the given msg using the clusterAccessOptions' IOStreams
+func (c *clusterAccessOptions) Print(msg string) {
+	osdctlutil.StreamPrint(c.IOStreams, msg)
+}
+
+// Errorln appends a newline then prints the given error msg using the clusterAccessOptions' IOStreams
+func (c *clusterAccessOptions) Errorln(msg string) {
+	osdctlutil.StreamErrorln(c.IOStreams, msg)
+}
+
+// Readln reads a single line of user input using the clusterAccessOptions' IOStreams. User input is returned with all
+// procceeding and following whitespace trimmed
+func (c *clusterAccessOptions) Readln() (string, error) {
+	in, err := osdctlutil.StreamRead(c.IOStreams, '\n')
+	return strings.TrimSpace(in), err
+}
+
+// Run executes the 'cluster' access subcommand
+func (c *clusterAccessOptions) Run(cmd *cobra.Command, args []string) error {
+	clusterIdentifier := args[0]
+	c.Println(fmt.Sprintf("Retrieving Kubeconfig for cluster '%s'", clusterIdentifier))
+
+	// Connect to ocm
+	conn := osdctlutil.CreateConnection()
+	defer func() {
+		cmdutil.CheckErr(conn.Close())
+	}()
+
+	cluster, err := osdctlutil.GetCluster(conn, clusterIdentifier)
+	if err != nil {
+		return err
+	}
+	c.Println(fmt.Sprintf("Internal Cluster ID: %s", cluster.ID()))
+
+	// Retrieve the kubeconfig secret from the cluster's namespace on hive
+	ns, err := getClusterNamespace(c.Client, cluster.ID())
+	if err != nil {
+		return err
+	}
+	c.Println(fmt.Sprintf("Cluster namespace: %s", ns.Name))
+
+	kubeconfigSecret, err := c.getKubeConfigSecret(ns)
+	if err != nil {
+		return err
+	}
+	c.Println(fmt.Sprintf("Kubeconfig Secret: %s", kubeconfigSecret.Name))
+
+	if cluster.AWS().PrivateLink() {
+		c.Println("")
+		c.Println("Cluster is PrivateLink, and is only accessible via a jump pod on Hive")
+		return c.createJumpPodAccess(cluster, kubeconfigSecret)
+	} else {
+		// Cluster is not PrivateLink - save kubeconfig locally
+		c.Println("")
+		c.Println("Cluster is accessible via a local Kubeconfig file")
+		return c.createLocalKubeconfigAccess(cluster, kubeconfigSecret)
+	}
+}
+
+// createJumpPodAccess grants access to a cluster by creating a pod for users to exec into
+func (c *clusterAccessOptions) createJumpPodAccess(cluster *clustersmgmtv1.Cluster, kubeconfigSecret corev1.Secret) error {
+	c.Println("Attempting to spin up a pod to use for access")
+
+	pod, err := c.createJumpPod(kubeconfigSecret, cluster.ID())
+	if err != nil {
+		c.Errorln("Failed to create pod")
+		return err
+	}
+
+	c.Println(fmt.Sprintf("Jump pod created. Waiting for it to start"))
+	c.Println("")
+
+	err = c.waitForJumpPod(pod, jumpPodPollInterval, jumpPodPollTimeout)
+	if err != nil {
+		c.Errorln("Timed out waiting for pod to start.")
+		c.Println(fmt.Sprintf("You can check the status of the pod using\n\n    oc describe pods %s -n %s\n", pod.Name, pod.Namespace))
+		c.Println("Once the pod is running:")
+	} else {
```

Comment:
else not needed

Reply:
The else is needed here to conditionally run the block